tests/test_edf.py

```diff
@@ -1029,3 +1030,38 @@ def test_get_starttime_from_file_with_reserved_field_indicating_edfplus_but_no_a
     edf._reserved = Edf.reserved.encode("EDF+C")
     edf.write(tmp_file)
     assert read_edf(tmp_file).starttime == starttime
+
+
+def test_resample_signal():
+    edf = Edf(
+        [
+            EdfSignal(np.arange(10), 10, digital_range=(0, 9), physical_range=(0, 9)),
+        ],
+    )
+    edf.signals[0].resample(5, np.arange(0, 10, 2))
+    assert edf.signals[0].sampling_frequency == 5
+    assert_array_equal(edf.signals[0].data, np.arange(0, 10, 2))
```

Comment:
For consistency with the existing tests, we could use `np.testing.assert_array_equal` here (with no explicit import)